relay-cardinality/src/limiter.rs

```diff
@@ -159,10 +161,19 @@ pub struct Accepted<T> {
 
 impl<T> Accepted<T> {
     fn new(source: Vec<T>, rejections: BTreeSet<usize>) -> Self {
-        Self {
-            source: source.into_iter(),
-            rejections: rejections.into_iter().peekable(),
-            next_index: 0,
+        if rejections.len() == source.len() {
```

Comment:
This checks whether both have the same length and assumes it's always 0. How does this code behave when `source.len() == 1` and `rejections.len() == 1`?

Reply:
When we have the same amount of items and rejections it means all items have been rejected and we have 0 items that are accepted -> short circuit with empty.